react/features/calendar-sync/middleware.js

```diff
@@ -262,27 +267,58 @@ function _parseCalendarEntry(event, knownDomains) {
  * @returns {void}
  */
 function _updateCalendarEntries(events) {
-    if (events && events.length) {
-        // eslint-disable-next-line no-invalid-this
-        const { dispatch, getState } = this;
-
-        const knownDomains = getState()['features/base/known-domains'];
-        const eventList = [];
-
-        const now = Date.now();
-
-        for (const event of events) {
-            const calendarEntry = _parseCalendarEntry(event, knownDomains);
+    if (!events || !events.length) {
+        return;
+    }
 
-            calendarEntry
-                && calendarEntry.endDate > now
-                && eventList.push(calendarEntry);
+    // eslint-disable-next-line no-invalid-this
+    const { dispatch, getState } = this;
+    const knownDomains = getState()['features/base/known-domains'];
+    const now = Date.now();
+    const entryMap = new Map();
+
+    for (const event of events) {
+        const entry = _parseCalendarEntry(event, knownDomains);
+
+        if (entry && entry.endDate > now) {
+            // As was stated above, we don't display subsequent occurrences of
+            // recurring events, and the repetitions of events coming from
+            // multiple calendars.
+            const key = md5.hex(JSON.stringify({
+
+                // Obviously, we want to display different conference/meetings
+                // URLs. URLs are the very reason why we implemented the feature
+                // calendar-sync in the first place.
+                u: entry.url,
+
+                // We probably want to display one and the same URL to people if
+                // they have it under different titles in their Calendar.
+                // Because maybe they remember the title of the meeting, not the
+                // URL so they expect to see the title without realizing that
+                // they have the same URL already under a different title.
+                t: entry.title,
+
+                // XXX Eventually, given that the URL and the title are the
+                // same, what sets one event apart from another is the start
+                // time of the day (note the use of toTimeString() bellow)! The
+                // day itself is not important because we don't want multiple
+                // occurrences of a recurring event or repetitions of an even
+                // from multiple calendars.
+                d: new Date(entry.startDate).toTimeString()
```

Comment:
@zbettenbuk, I've left comments here on every key property in order to clarify to readers how I figured what goes into key and to attract the attention to the distinction between time and date + time.